dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/testing/FdcProcessSpy.java

```diff
@@ -0,0 +1,104 @@
+
+/**
+ * See {@link SpyFactory} for usage details.
+ */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Builder
+@Getter
+public class FdcProcessSpy {
+  private final Set<Integer> activeIds;         // Returned from maat-api by FdcClient.getContributions("ACTIVE")
+  @Singular
+  private final Set<Integer> sentIds;           // Sent to the DRC by DrcClient.sendContributionUpdate(...)
+  private final Set<Integer> xmlCcIds;          // Sent to maat-api by FdcClient.updateContribution(...)
+  private final int recordsSent;                //  "    "    "
+  private final String xmlContent;              //  "    "    "
+  private final String xmlFileName;             //  "    "    "
+  private final Integer xmlFileResult;          // Returned from maat-api by FdcClient.updateContribution(...)
+  private final FdcGlobalUpdateResponse globalUpdateResponse;
+
+  private static FdcProcessSpyBuilder builder() {
+    throw new UnsupportedOperationException("Call SpyFactory.newFdcProcessSpyBuilder instead");
+  }
+
+  public static class FdcProcessSpyBuilder {
+    private final FdcClient FdcClientSpy;
+    private final DrcClient drcClientSpy;
+
+    private FdcProcessSpyBuilder() {
+      throw new UnsupportedOperationException("Call SpyFactory.newFdcProcessSpyBuilder instead");
+    }
+
+    FdcProcessSpyBuilder(final FdcClient FdcClientSpy, final DrcClient drcClientSpy) {
+      this.FdcClientSpy = FdcClientSpy;
+      this.drcClientSpy = drcClientSpy;
+    }
+
+    public FdcProcessSpyBuilder traceAndFilterGetFdcContributions(final Set<Integer> updatedIds) {
+      final var idSet = Set.copyOf(updatedIds); // defensive copy
+      doAnswer(invocation -> {
+        var result = (FdcContributionsResponse) mockingDetails(FdcClientSpy).getMockCreationSettings().getDefaultAnswer().answer(invocation);
+        result.setFdcContributions(result.getFdcContributions().stream().filter(fdcContributionEntry -> idSet.contains(fdcContributionEntry.getId())).toList());
+        activeIds(result.getFdcContributions().stream().map(FdcContributionEntry::getId).collect(Collectors.toSet()));
+        return result;
+      }).when(FdcClientSpy).getFdcContributions(REQUESTED_STATUS);
+      return this;
+    }
+
+    public FdcProcessSpyBuilder traceAndStubSendFdcUpdate(final Predicate<Integer> stubResults) {
+      doAnswer(invocation -> {
+        var fdcId = ((SendFdcFileDataToDrcRequest) invocation.getArgument(0)).getFdcId();
+        sentId(fdcId);
+        return stubResults.test(fdcId);
+      }).when(drcClientSpy).sendFdcUpdate(any());
+      return this;
+    }
+
+    public FdcProcessSpyBuilder traceUpdateFdcs() {
+      doAnswer(invocation -> {
+        final var data = (FdcUpdateRequest) invocation.getArgument(0);
+        xmlCcIds(data.getFdcIds().stream().map(Integer::valueOf).collect(Collectors.toSet()));
+        recordsSent(data.getRecordsSent());
+        xmlContent(data.getXmlContent());
+        xmlFileName(data.getXmlFileName());
```

Comment:
I might be guilty of not doing this myself, but you could chain these if it made things more readable (not sure if it does or not - your call):

```
this.xmlCcIds(...)
    .recordsSent(...)
    .xmlContent(...)
    .xmlFileName(...)
```